### ISSUE TITLE
localhost keycloak broken

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/PermaplanT#416]
```
AUTH_DISCOVERY_URI=https://auth.permaplant.net/realms/PermaplanT/.well-known/openid-configuration
AUTH_CLIENT_ID=localhost
```

leads to an error when logging in `https://auth.permaplant.net/realms/PermaplanT/protocol/openid-connect/auth?client_id=localhost&redirect_uri=http%3A%2F%2Flocalhost%3A5173%2Fmap&response_type=code&scope=openid&state=...&code_challenge=...-...&code_challenge_method=S256&response_mode=query`

```
We are sorry...
Invalid parameter: redirect_uri
```

It happened on #404 but is probably unrelated to that.

**Comment by absurd-turtle** [comment]
The login works for me on the base uri: http://localhost:5173

**Comment by markus2330** [comment]
Thx, this can be used as workaround!

**Comment by lukashartl** [comment]
According to the docs:

https://github.com/ElektraInitiative/PermaplanT/blob/a488fb88e200f473eaf57bde2ea169c93fac6eea/doc/setups/keycloak/README.md?plain=1#L26-L28

```json
{
  "clientId" : "localhost",
  "name" : "localhost",
  "rootUrl" : "http://localhost:5173/",
  "adminUrl" : "http://localhost:5173/",
  "redirectUris" : [ "/*" ],
  "webOrigins" : [ "+" ],
  "publicClient" : true
}
```

so those seem to match.

Maybe we need to add the full `rootUrl` to `redirectUris`.

**Comment by kitzbergerg** [comment]
> According to the docs:
> 
> https://github.com/ElektraInitiative/PermaplanT/blob/a488fb88e200f473eaf57bde2ea169c93fac6eea/doc/setups/keycloak/README.md?plain=1#L26-L28
> 
> ```json
> {
>   "clientId" : "localhost",
>   "name" : "localhost",
>   "rootUrl" : "http://localhost:5173/",
>   "adminUrl" : "http://localhost:5173/",
>   "redirectUris" : [ "/*" ],
>   "webOrigins" : [ "+" ],
>   "publicClient" : true
> }
> ```
> 
> so those seem to match.
> 
> Maybe we need to add the full `rootUrl` to `redirectUris`.

You have a slash at the end of `rootUrl`. That might make a difference.

**Comment by lukashartl** [comment]
You are right! I totally missed the trailing slash! I'll take a look at this and other clients as well (e.g. #383).
